pkg/brownfield/routing_rules_test.go

```diff
@@ -101,4 +102,104 @@ var _ = Describe("Test blacklist request routing rules", func() {
 			Expect(blacklisted).To(ContainElement(rulePathBased3))
 		})
 	})
+
+	Context("Test MergeRules", func() {
+		It("should merge correctly when same set of routing rules need to be merged", func() {
+			Expect(len(*appGw.RequestRoutingRules)).To(Equal(5))
+			Expect(len(*appGw.URLPathMaps)).To(Equal(4))
+
+			requestRoutingRules := MergeRules(&appGw, *appGw.RequestRoutingRules, *appGw.RequestRoutingRules)
+
+			// No change
+			Expect(len(requestRoutingRules)).To(Equal(5))
+			Expect(len(*appGw.URLPathMaps)).To(Equal(4))
+
+			pathBasedRuleCount := 0
+			basicRuleCount := 0
+			for _, rule := range *appGw.RequestRoutingRules {
+				if rule.RuleType == n.ApplicationGatewayRequestRoutingRuleTypePathBasedRouting {
+					pathBasedRuleCount++
+					Expect(lookupPathMap(appGw.URLPathMaps, rule.URLPathMap.ID)).ToNot(BeNil())
+				}
+
+				if rule.RuleType == n.ApplicationGatewayRequestRoutingRuleTypeBasic {
+					basicRuleCount++
+				}
+			}
+
+			Expect(pathBasedRuleCount).To(Equal(3))
+			Expect(basicRuleCount).To(Equal(2))
+		})
+
+		It("should merge correctly when different set of routing rules need to be merged", func() {
+			requestRoutingRules := MergeRules(
+				&appGw,
+				[]n.ApplicationGatewayRequestRoutingRule{
+					ruleBasic,
+					ruleDefault,
+					rulePathBased1,
+				},
+				[]n.ApplicationGatewayRequestRoutingRule{
+					rulePathBased2,
+					rulePathBased3,
+				})
+
+			Expect(len(requestRoutingRules)).To(Equal(5))
+			Expect(len(*appGw.URLPathMaps)).To(Equal(4))
```

Comment:
Is it 4 before doing the merge?

Reply:
yes and it should not decrease as we should retain all the url path maps